app/models/appointment.rb

```diff
@@ -1,5 +1,23 @@
 class Appointment < ActiveRecord::Base
 	belongs_to :doctor
 	belongs_to :patient
-	validates_presence_of :doctor
+
+  def self.overload(doctor_id,schedule)
+    appointment_count = Appointment.where(doctor_id: doctor_id, status:'Approved', schedule: schedule ).count
+    if appointment_count < 4
+      return false
+    else
+      return true
+    end
+  end
+
+  def self.check_role(role,appointment)
+    if role == 'admin'
+        appointment.status = 'Pending'
+        appointment.save!
+    elsif role == 'superadmin'
+        appointment.status = 'Approved'
```

Comment:
Why explicitly assign status for a superadmin? This would break if the superadmin edits the appointment and doesn't change the status, i.e. the status is still pending.